base/ltfilehook.dtx

```diff
@@ -1117,6 +1128,32 @@
 %    \end{macrocode}
 %
 %    \begin{macrocode}
+%<latexrelease>\IncludeInRelease{2021/06/01}%
+%<latexrelease>          {\set@curr@file}{Setting current file name}%
+%<latexrelease>\def\set@curr@file#1{%
+%<latexrelease>  \begingroup
+%<latexrelease>    \escapechar\m@ne
+%<latexrelease>    \let\protect\string
+%<latexrelease>    \edef~{\string~}%
+%<latexrelease>    \expandafter\def\csname\expandafter\endcsname
+%<latexrelease>      \expandafter{\csname\endcsname}%
+%<latexrelease>    \@expl@@@filehook@if@no@extension@@nTF{#1}%
+%<latexrelease>      {\@tempswatrue}{\@tempswafalse}%
+%<latexrelease>    \@kernel@make@file@csname\@curr@file
+%<latexrelease>      \@expl@@@filehook@resolve@file@subst@@w {#1}%
+%<latexrelease>    \@expl@@@filehook@if@file@replaced@@TF
+%<latexrelease>      {\@kernel@make@file@csname\@curr@file@reqd
+%<latexrelease>         \@expl@@@filehook@normalize@file@name@@w{#1}%
+%<latexrelease>       \if@tempswa \@expl@@@filehook@drop@extension@@N\@curr@file@reqd \fi}%
+%<latexrelease>      {\if@tempswa \@expl@@@filehook@drop@extension@@N\@curr@file \fi
+%<latexrelease>       \global\let\@curr@file@reqd\@curr@file}%
+%<latexrelease>    \@expl@@@filehook@clear@replacement@flag@@
+%<latexrelease>  \endgroup}
+%<latexrelease>\let\set@curr@file@nosearch\set@curr@file
```

Comment:
I guess this is needed because if the kernel is rolled back graphics still uses that command. However, there should probably be a remark to that effect. Furthermore, what happens if the kernel is rolled back to some earlier state? Then the definition of `\setcurrfile`is altered back further (but  this guy stays at its definition and thus I'm failry sure chaos is the result ... or am I overlooking something?). In other words I guess that `\let` is also needed in all earlier parts.

Reply:
> I guess this is needed because if the kernel is rolled back graphics still uses that command.

Bah, yes, `graphics.sty` doesn't roll back at all.

> Furthermore, what happens if the kernel is rolled back to some earlier state?

The kernel will use whatever definition of `\set@curr@file` was current at that time, and `graphics` should use that same `\set@curr@file` (but now named `@nosearch`).

> In other words I guess that `\let` is also needed in all earlier parts.

Yes, I think that should be enough.  I'll do that plus some tests in a bit.

Reply:
That's why I think we should add rollback tests both to base and to tools back to the last 4-5 main releases (and in tools load our packages and check that they actually remain functional  (as far as we can easily check). No of that will be perfect but better than nothing.

Reply:
and sorry to spoil it :-)

Reply:
I ended up doing something else.  I think it is better, as far as possible, to have `latexrelease` roll back as close to _real_ rollback as possible, so instead of making `\set@curr@file@nosearch` a copy of `\set@curr@file` (which would also be a lie, because it _would_ search), I changed `graphics` to use `\Gin@set@curr@file`, and within `graphics` itself I pick up the proper definition for that internal macro, depending on what is available.  Indirectly this also supports plain, and whatever change `latexrelease` ends up doing to `\set@curr@file`, so it seems like a good solution to me.  Of course, I still have to manually undefine `\set@curr@file@nosearch` in every prior `\IncludeInRelease` block.